### PR TITLE
Upgrade `react-grid-system` to v^8.1.4

### PR DESCRIPTION
**Pull request by NestorEncinas**
JIRA: FET-1062

---

Supported PR commands:

| Command                  | Description             |
| ------------------------ | ----------------------- |
| `ok to test`             | Re-run standard checks  |
| `extended test`          | BackstopJS tests        |
| `extended check sonar`   | SonarQube tests         |
| `extended check cypress` | Cypress E2E tests       |
| `extended check plugins` | Dashboard plugins tests |

---

# PR Checklist

-   [ ] commit messages adhere to the [commit message guidelines](https://github.com/gooddata/gooddata-ui-sdk/blob/master/docs/contributing.md#what-should-the-commits-look-like)
-   [ ] review was done by a Code owner [if necessary](https://github.com/gooddata/gooddata-ui-sdk/blob/master/docs/contributing.md#how-do-i-tell-if-my-pull-request-needs-approval-by-a-code-owner) (if you think it is not necessary, explain the reasoning in the description or in a comment)
-   [ ] `check` passes
-   [ ] `check-extended` passes
-   [ ] `check-extended-cypress` passes
-   [ ] `rush change` [was run if applicable](https://github.com/gooddata/gooddata-ui-sdk/blob/master/docs/contributing.md#how-do-i-describe-my-changes-for-the-changelog)

### CODE REVIEW

**Comment by gdgate** [comment]
Build succeeded (check pipeline).

- https://checklist.intgdc.com/job/compliance/job/gooddata-ui-sdk-oss-check/8683/ : SUCCESS in 8s
- https://checklist.intgdc.com/job/gooddata-ui-sdk/job/gooddata-ui-sdk-unit-tests/10543/ : SUCCESS in 21m 01s

**Comment by NestorEncinas** [comment]
extended test

**Comment by gdgate** [comment]
Build succeeded (check-extended pipeline).

- https://checklist.intgdc.com/job/compliance/job/gooddata-ui-sdk-oss-check/8684/ : SUCCESS in 7s
- https://checklist.intgdc.com/job/gooddata-ui-sdk/job/gooddata-ui-sdk-backstop-tests/1694/ : SUCCESS in 27m 37s

**Comment by NestorEncinas** [comment]
extended check cypress

**Comment by gdgate** [comment]
Build failed (check-extended pipeline).

- https://checklist.intgdc.com/job/compliance/job/gooddata-ui-sdk-oss-check/8685/ : SUCCESS in 6s
- https://checklist.intgdc.com/job/gooddata-ui-sdk/job/gooddata-ui-sdk-cypress-tests/241/ : UNSTABLE in 12m 28s

**Comment by NestorEncinas** [comment]
extended check plugins

**Comment by gdgate** [comment]
Build succeeded (check-extended pipeline).

- https://checklist.intgdc.com/job/gooddata-ui-sdk/job/gooddata-ui-sdk-dashboard-plugin-tests/24/ : SUCCESS in 10m 34s
- https://checklist.intgdc.com/job/compliance/job/gooddata-ui-sdk-oss-check/8688/ : SUCCESS in 6s

File: libs/sdk-ui-dashboard/src/presentation/layout/DefaultDashboardLayoutRenderer/interfaces.ts [hunk 1]
```diff
@@ -19,7 +20,7 @@ export type IDashboardLayoutSectionKeyGetterProps<TWidget = IDashboardWidget> =
     /**
      * Current screen type with respect to the set breakpoints.
      */
-    screen: ScreenSize;
+    screen: ScreenClass;
```
Comment (ivan-nejezchleb) [inline libs/sdk-ui-dashboard/src/presentation/layout/DefaultDashboardLayoutRenderer/interfaces.ts:23]: I am not sure if we can change this type of public interface, before it used our internal type. 
Should not we do some conversion from 
ScreenSize from "@gooddata/sdk-model"
to 
import { ScreenClass } from "react-grid-system";
internally?

**Comment by gdgate** [comment]
Build succeeded (check pipeline).

- https://checklist.intgdc.com/job/compliance/job/gooddata-ui-sdk-oss-check/8691/ : SUCCESS in 7s
- https://checklist.intgdc.com/job/gooddata-ui-sdk/job/gooddata-ui-sdk-unit-tests/10546/ : SUCCESS in 19m 39s

**Comment by gdgate** [comment]
Build succeeded (gate pipeline).

- https://checklist.intgdc.com/job/compliance/job/gooddata-ui-sdk-fossa-scanning-zuul-el7/2500/ : SUCCESS in 4m 52s (non-voting)
- https://checklist.intgdc.com/job/compliance/job/gooddata-ui-sdk-merge-compliance-check/2796/ : SUCCESS in 21s
- https://checklist.intgdc.com/job/compliance/job/gooddata-ui-sdk-sonarqube-scan-zuul-gate/1624/ : SUCCESS in 16m 49s
- https://checklist.intgdc.com/job/gooddata-ui-sdk/job/gooddata-ui-sdk-unit-tests/10547/ : SUCCESS in 19m 36s

**Pull request merged**